pages/profile/read/views/index.jsx

```diff
@@ -85,61 +91,62 @@ const Index = ({
-                      !isEmpty(roles) && (
-                        <Fragment>
-                          <dt><Snippet>roles</Snippet></dt>
-                          <dd>
-                            <ul>
-                              {
-                                roles.map(({ type, id }) =>
-                                  <li key={id}>{defineValue(type.toUpperCase())}</li>
-                                )
-                              }
-                            </ul>
-
-                          </dd>
-                        </Fragment>
-                      )
-                    }
-                    {
-                      !isEmpty(premises) && (
-                        <Fragment>
-                          <dt><Snippet>premises</Snippet></dt>
-                          <dd>
-                            <ul>
-                              {
-                                map(premises, (p, site) =>
-                                  <li key={site}>
-                                    {site}
-                                    {
-                                      map(p, (names, area) =>
-                                        <Fragment key={area}>
-                                          {
-                                            area !== 'null'
-                                              ? <Fragment>, {area} - <br /></Fragment>
-                                              : <Fragment> - <br /></Fragment>
-                                          }
-                                          {
-                                            names.join(', ')
-                                          }
-                                        </Fragment>
-                                      )
-                                    }
-                                  </li>
-                                )
-                              }
-                            </ul>
-                          </dd>
-                        </Fragment>
-                      )
-                    }
-                  </dl>
-                </ExpandingPanel>
-              )
+              <ExpandingPanel title={<Snippet>responsibilities.title</Snippet>}>
+                <dl className="inline">
+                  {
+                    <Fragment>
+                      <dt><Snippet>responsibilities.roles</Snippet></dt>
+                      <dd>
+                        { !isEmpty(roles) &&
+                          <ul>
+                            {
+                              roles.map(({ type, id }) =>
+                                <li key={id}>{defineValue(type.toUpperCase())}</li>
+                              )
+                            }
+                          </ul>
+                        }
+                        { isEmpty(roles) &&
+                          <Snippet>responsibilities.noRoles</Snippet>
+                        }
+                        <RoleApply />
```

Comment:
This should only be visible if the logged in user has permission to edit roles.

Reply:
Actually, I see the component itself handles that. That's probably a better way of doing it 👍